### PR TITLE
AliasAnalysis handles basic `Set` (no permute).

### PR DESCRIPTION
**Pull request by wujingyue**

### CODE REVIEW

**Comment by wujingyue** [comment]
!build

**Comment by wujingyue** [comment]
Some unit tests are failing because they assume `consumer=set(producer)` triggers the pointwise scheduler, which is no longer true after this PR. Fixing those tests...

**Comment by wujingyue** [comment]
!build

File: test/test_alias.cpp [hunk 1]
```diff
@@ -108,6 +108,29 @@ TEST_F(AliasAnalysisTest, View_MergeNonContiguous) {
...
+TEST_F(AliasAnalysisTest, Set) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  const std::vector<int64_t> in_shape({2, 3, 5});
+
+  TensorView* in = makeContigConcreteTensor(in_shape);
+  fusion.addInput(in);
+  TensorView* out = set(in);
+  fusion.addOutput(out);
```
Comment (jjsjann123) [inline test/test_alias.cpp:120]: nitpick: can we also add a test where we have output stride set that's not consistent/compatible with the input stride and would block aliasing?

**Review by jjsjann123 — APPROVED** [review]
Thanks for cleaning-up and separating the tests. :bow:

File: test/test_alias.cpp (hunk 1 above)
Comment (wujingyue) [reply test/test_alias.cpp:118]: Done. Note that AliasAnalysis itself still produces the alias with a preferred allocation domain; it's MarkAliasPass that checks the allocation domain is compatible.

**Comment by wujingyue** [comment]
!build

**Comment by jjsjann123** [comment]
:shipit:

**Pull request merged**